Test Snyk (#374)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/webhook/examples/snyk.md

:::

Done! Any vulnerability detected on your source code will trigger a webhook event to the webhook URL provided by Port. Port will parse the events according to the mapping and update the catalog entities accordingly.

> ## Let's Test It

Reviewer: ```suggestion
## Test the webhook
```

> In this section, we'll explore the webhook event data that is received from Snyk whenever a code vulnerability is added or removed from your repository. We'll also delve into how the entity is finally created in Port by using the webhook configuration.

Reviewer: ```suggestion
This section includes a sample webhook event sent from Snyk when a code vulnerability is added or removed. In addition, it also includes the entity created from the event based on the webhook configuration provided in the previous section.
```

### Payload

> Below is an example of the payload structure sent to the webhook URL after a code vulnerability is detected:

Reviewer: ```suggestion
Here is an example of the payload structure sent to the webhook URL when a code vulnerability is detected:
```

        "identifiers": {
          "CVE": [],
          "CWE": ["CWE-310"]
        },
        "credit": ["Felix"],
        "exploitMaturity": "proof-of-concept",
        "semver": {
          "vulnerable": ["<5.0.0"]
        },
        "publicationTime": "2020-07-01T15:32:37.471739Z",
        "disclosureTime": "2020-01-13T13:45:13Z",
        "CVSSv3": "CVSS:3.1/AV:N/AC:H/PR:N/UI:N/S:U/C:N/I:H/A:N/E:P/RL:O/RC:R",
        "cvssScore": 5.9,
        "cvssDetails": [],
        "functions": [],
        "language": "js",
        "patches": [],
        "nearestFixedInVersion": "",
        "isMaliciousPackage": false
      },
      "isPatched": false,
      "isIgnored": false,
      "fixInfo": {
        "isUpgradable": true,
        "isPinnable": false,
        "isPatchable": false,
        "isFixable": true,
        "isPartiallyFixable": true,
        "nearestFixedInVersion": "",
        "fixedIn": ["5.0.0"]
      }
    }
  ]
}
```

</details>

### Mapping Result

> Using the mappings defined in the webhook configuration, Port will extract the necessary properties from the Snyk webhook payload and use the resulting data to create the vulnerability entities. In this case, because there are two sets of data in the `newIssues` array property (check the above JSON data), Port will create two vulnerability entities. Here is an example of one of the mappings:

Reviewer: ```suggestion
The combination of the sample payload and the webhook configuration generate the following Port entities (in the sample above, multiple entities will be generated because the `newIssues` array contains multiple objects):
```
